File: interfaces/src/main/java/com/nerdscorner/mvplib/interfaces/presenter/BaseActivityPresenter.java
package com.nerdscorner.mvplib.interfaces.presenter;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.Menu;
import android.view.MenuItem;

import com.nerdscorner.mvplib.interfaces.model.InterfacesModelInterface;
import com.nerdscorner.mvplib.interfaces.view.InterfacesViewInterface;

public class BaseActivityPresenter<V extends InterfacesViewInterface, M extends InterfacesModelInterface> extends BaseInterfacesPresenter<V, M> {

    public BaseActivityPresenter(@NonNull V view, @NonNull M model) {
        super(view, model);
    }

    @Override
    public boolean onBackPressed() {
        return false;
    }

    public boolean onCreateOptionsMenu(Menu menu) {
        return false;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        return false;
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
    }

    @Override
    public void onRestoreInstanceState(Bundle savedInstanceState) {
    }

    public void onDestroy() {
        view.onDestroy();
        model.onDestroy();
    }
}

File: events/src/main/java/com/github/marcherdiego/mvp/events/presenter/BaseActivityPresenter.kt
package com.github.marcherdiego.mvp.events.presenter

import android.content.Intent
import android.os.Bundle
import com.github.marcherdiego.mvp.events.bus.Bus
import com.github.marcherdiego.mvp.events.model.BaseEventsModel
import com.github.marcherdiego.mvp.events.view.BaseActivityView

open class BaseActivityPresenter<V : BaseActivityView, M : BaseEventsModel>
@JvmOverloads constructor(view: V, model: M, internal val bus: Bus = Bus.defaultBus) : BasePresenter<V, M>(view, model) {

    init {
        view.setBus(bus)
        model.setBus(bus)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {}

    override fun onRestoreInstanceState(savedInstanceState: Bundle?) {}

    open fun onDestroy() {}
}
